packages/airnode-node/src/workers/local-gateways/server.ts

```diff
@@ -0,0 +1,185 @@
+import { logger } from '@api3/airnode-utilities';
+import express, { Request, Response } from 'express';
+import { z } from 'zod';
+import bodyParser from 'body-parser';
+import { VerificationResult, verifyHttpRequest, verifyHttpSignedDataRequest, verifyRequestOrigin } from './validation';
+import { Config, EnabledGatewaySchema, LocalProvider } from '../../config';
+import { processHttpRequest, processHttpSignedDataRequest } from '../../handlers';
+
+type GatewayName = 'httpGateway' | 'httpSignedDataGateway';
+
+function verifyApiKey(config: Config, req: Request, gatewayName: GatewayName): VerificationResult<{}> {
+  const apiKey = req.header('x-api-key');
+  const gateway = config.nodeSettings[gatewayName];
+
+  if (!apiKey || !gateway.enabled || apiKey !== gateway.apiKey) {
+    // Mimics the behavior of AWS HTTP Gateway
+    return { success: false, statusCode: 403, error: { message: 'Forbidden' } };
+  }
+
+  return { success: true };
+}
+
+// We do not want to enable ".strict()" - we want to allow extra fields in the request body
+const httpRequestBodySchema = z.object({
+  parameters: z.any(), // Parameter validation is performed later
+});
+// We do not want to enable ".strict()" - we want to allow extra fields in the request body
+const httpSignedDataBodySchema = z.object({
+  encodedParameters: z.string(),
+});
+const DEFAULT_PORT = 3000;
+
+export function getGatewaysUrl(port: number = DEFAULT_PORT, path?: string) {
+  const base = `http://localhost:${port || DEFAULT_PORT}`;
+  if (!path) return base;
+  return base + (path.startsWith('/') ? path : `/${path}`);
+}
+
+export const HTTP_SIGNED_DATA_BASE_PATH = '/http-signed-data';
+export const HTTP_BASE_PATH = '/http-data';
+
+export function startGatewayServer(config: Config, enabledGateways: GatewayName[]) {
+  if (enabledGateways.length === 0) {
+    logger.log('Not starting API gateway server because there is no gateway enabled.');
+    return;
+  }
+
+  const app = express();
+  app.use(bodyParser.json());
+  const cloudProviderSettings = config.nodeSettings.cloudProvider as LocalProvider;
+  const port = cloudProviderSettings.gatewayServerPort ?? DEFAULT_PORT;
+
+  if (enabledGateways.includes('httpSignedDataGateway')) {
+    const httpSignedDataGatewayPath = `${HTTP_SIGNED_DATA_BASE_PATH}/:endpointId`;
+    const httpSignedDataRequestHandler = async function (req: Request, res: Response) {
+      logger.log(`Received request for http signed data`);
+
+      const originVerification = verifyRequestOrigin(
+        (config.nodeSettings.httpSignedDataGateway as EnabledGatewaySchema).corsOrigins,
+        req.headers.origin
+      );
+      if (req.method === 'OPTIONS') {
+        if (!originVerification.success) {
+          res.status(400).send(originVerification.error);
+          return;
+        }
+
+        // Set headers for the responses
+        res.set(originVerification.headers).status(204).send('');
+        return;
+      }
+      res.set(originVerification.headers);
```

Comment:
```suggestion
        res.set(originVerification.headers).status(204).send('');
        return;
      }
       // Set headers for the responses
      res.set(originVerification.headers);
```
Nit: to move the comment down

Reply:
I enriched the comment a bit. LMK if it makes sense.

Reply:
Yep looks good :+1: